esp_at_lib/src/api/esp_netconn.c

```diff
@@ -392,6 +392,45 @@ esp_netconn_connect(esp_netconn_p nc, const char* host, esp_port_t port) {
     return res;
 }
 
+/**
+ * \brief           Connect to server as client, allow keep-alive option
+ * \param[in]       nc: Netconn handle
+ * \param[in]       host: Pointer to host, such as domain name or IP address in string format
+ * \param[in]       port: Target port to use
+ * \param[in]       keep_alive: keep alive period seconds
+ * \param[in]       local_ip: local ip in connected command
+ * \return          \ref espOK if successfully connected, member of \ref espr_t otherwise
+ */
+espr_t
+esp_netconn_connect_ex(esp_netconn_p nc, const char* host, esp_port_t port, uint16_t keep_alive, const char* local_ip, esp_port_t local_port, uint8_t mode) {
+    esp_conn_start_t cs[1] = {0};
```

Comment:
Why did you decice for an array with 1 element instead of classic variable?

Reply:
This we need to re-think and create a esp_netconn_start_t (or similar) structure to add all parameters there. This is to ensure future features could be easily added w/o breaking compatibility. I will do this.